pkg/maintainer/spv/spv.go

```diff
@@ -59,8 +63,271 @@ func (sm *spvMaintainer) startControlLoop(ctx context.Context) {
 func (sm *spvMaintainer) maintainSpv(ctx context.Context) error {
 	logger.Infof("Maintaining SPV proof...")
 
-	// TODO: Implement. For now, just wait.
-	<-ctx.Done()
+	for {
+		if err := sm.proveDepositSweepTransactions(); err != nil {
+			return fmt.Errorf(
+				"error while proving deposit sweep transactions: [%v]",
+				err,
+			)
+		}
+
+		// TODO: Add proving of other type of SPV transactions: redemption
+		// transactions, moving funds transaction, etc.
+
+		select {
+		case <-time.After(sm.config.IdleBackOffTime):
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+}
+
+func (sm *spvMaintainer) proveDepositSweepTransactions() error {
+	depositSweepTransactions, err := sm.getUnprovenDepositSweepTransactions()
+	if err != nil {
+		return fmt.Errorf(
+			"failed to get unproven deposit sweep transactions: [%v]",
+			err,
+		)
+	}
+
+	// TODO: Consider handling a situation in which the block headers in the
+	//       proof span multiple Bitcoin difficulty epochs.
```

Comment:
And if it failed on-chain, the maintainer would keep trying to submit it anyway on each round of processing.
It would lead to unnecessary burning of funds from the `ReimbursementPool`.

Reply:
I'll do the TODO in a follow-up PR.

Reply:
The followup PR: https://github.com/keep-network/keep-core/pull/3667.